docs/mkdocs/docs/technical/contributing.md

```diff
@@ -332,3 +332,80 @@ To override that, use one of the following CMake variables*:
 
    (Only) `Python_ROOT_DIR` can also be set as an environment variable.
    Setting the others in the environment might have no effect.)
+
+Development Guidelines
+======================
+
+ArcticDB has a lot of write-time options to configure how the data is stored on disk.
+When adding new features, it is important to test not only the simplest case, but also how the new feature interacts with these various options.
+This section serves as a guide as to what should be considered when adding a new feature.1
+
+## Backwards compatibility
+
+### Data stored on disk
+
+Due to ArcticDB being a purely client-side database, it is possible for data to be written by clients that are of later versions than clients that need to read the same data.
+If the change is breaking, such that older clients will not be able to read the data, then this should be clearly documented both in the PR and in the online documentation.
+If possible, a version number should also be added such that future changes in the same area will display a more clear error message.1
+
+### API
+
+Any changes to the API (including when exceptions are thrown and the type of the exception) must be weighed up against the change breaking behaviour for existing users.
+This is particularly true of the `NativeVersionStore` API, as this has many users inside Man Group.
+
+### Snapshots
```

Comment:
What is a snapshot?

Reply:
https://docs.arcticdb.io/tutorials/snapshots/

https://docs.arcticdb.io/api/library#arcticdb.version_store.library.Library.snapshot

Reply:
Might be worth linking here actually.